ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -545,6 +556,8 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
 
         heartbeater.cancel();
 
+        failAfterCompactionIfSetForTest();
```

Comment:
instead of introducing this test only code here, you could use Mockito.spy in the test and override verifyTableIdHasNotChanged to always throw exception, that would have the same effect (probably you need the change the visibility to protected)

Reply:
+1